packages/portal/src/mixins/klaro.js

```diff
@@ -0,0 +1,123 @@
+export const version = '0.7.18';
+
+export default {
+  data() {
+    return {
+      cookieConsentRequired: false,
+      klaro: null,
+      klaroHeadScript: { src: `https://cdn.jsdelivr.net/npm/klaro@${version}/dist/klaro-no-css.js`, defer: true },
+      toastBottomOffset: '20px'
+    };
+  },
+
+  watch: {
+    '$i18n.locale': 'renderKlaro'
+  },
+
+  mounted() {
+    if (!this.klaro) {
+      this.klaro = window.klaro;
+    }
+
+    // If Matomo plugin is installed, wait for Matomo to load, but still render
+    // Klaro if it fails to.
+    const renderKlaroAfter = this.$waitForMatomo ? this.$waitForMatomo() : Promise.resolve();
+    renderKlaroAfter.catch(() => {}).finally(this.renderKlaro);
+  },
+
+  methods: {
+    renderKlaro() {
+      if (this.klaro) {
+        const config = this.klaroConfig();
+        const manager = this.klaro.getManager(config);
+
+        this.cookieConsentRequired = !manager.confirmed;
+
+        this.klaro.render(config, true);
+        manager.watch({ update: this.watchKlaroManagerUpdate });
+
+        setTimeout(() => {
+          this.setToastBottomOffset();
+        }, 100);
+      }
+    },
+
+    setToastBottomOffset() {
+      const cookieNoticeHeight = document.getElementsByClassName('cookie-notice')[0]?.offsetHeight;
+      this.toastBottomOffset = cookieNoticeHeight ? `${cookieNoticeHeight + 40}px` : '20px';
+    },
+
+    watchKlaroManagerUpdate(manager, eventType, data) {
+      let eventName;
+
+      if (eventType === 'saveConsents') {
+        eventName = {
+          accept: 'Okay/Accept all',
+          decline: 'Decline',
+          save: 'Accept selected'
+        }[data.type];
+      }
+
+      eventName && this.trackKlaroClickEvent(eventName);
+
+      setTimeout(() => {
+        this.setToastBottomOffset();
+      }, 10);
+    },
+
+    trackKlaroClickEvent(eventName) {
+      this.$matomo?.trackEvent('Klaro', 'Clicked', eventName);
+    },
+
+    klaroConfig() {
+      // TODO: uncomment when we have translations
+      // const locale = $i18n.locale;
+      // TODO: remove when we have translations
```

Comment:
Are these TODO's still relevant? I see the cookie banner is translated.
I'm not sure how it picks up on the translations though as the locale is still hard coded to 'en' below.

Reply:
TODOs no longer relevant. It was translating anyway because $i18n was translating per the user's UI locale and supplying that to Klaro, just with the `en` key.